lib/puppet/provider/cisco_ospf_area/cisco.rb

```diff
@@ -138,6 +151,48 @@ def properties_set(new_area=false)
           @nu.respond_to?("#{prop}=")
       end
     end
+    # custom setters which require one-shot multi-param setters
+    nssa_set
+  end
+
+  def nssa_set
+    attrs = {}
+    vars = [
+      :nssa,
+      :nssa_route_map,
+      :nssa_default_originate,
+      :nssa_no_redistribution,
+      :nssa_no_summary,
+    ]
+    return unless vars.any? { |p| @property_flush.key?(p) }
+    # At least one var has changed, get all vals from manifest
+    attrs = {}
+    vars.each do |p|
+      if @resource[p] == :default
+        attrs[p] = @nu.send("default_#{p}")
+      else
+        attrs[p] = @resource[p]
+        attrs[p] = PuppetX::Cisco::Utils.bool_sym_to_s(attrs[p])
+      end
+    end
+    hash = {}
+    if attrs[:nssa]
+      hash[:nssa] = attrs[:nssa]
+    else
+      @nu.nssa_set(hash)
+      return
+    end
+    if attrs[:nssa_route_map] == '' || attrs[:nssa_route_map].nil?
+      hash[:route_map] = ''
+      hash[:rm] = ''
+    else
+      hash[:route_map] = 'route-map'
+      hash[:rm] = attrs[:nssa_route_map]
+    end
+    hash[:default_information_originate] = attrs[:nssa_default_originate] ? 'default-information-originate' : ''
+    hash[:no_redistribution] = attrs[:nssa_no_redistribution] ? 'no-redistribution' : ''
+    hash[:no_summary] = attrs[:nssa_no_summary] ? 'no-summary' : ''
+    @nu.nssa_set(hash)
   end
```

Comment:
This entire method looks way more complicated than it needs to be. We should probably have a phone call to discuss. 

I would expect to see something like this instead:

```
  def nssa_set
    attrs = {}
    vars = [
      :nssa,
      :nssa_route_map,
      :nssa_default_originate,
      :nssa_no_redistribution,
      :nssa_no_summary,
    ]
    if vars.any? { |p| @property_flush.key?(p) }
      # At least one var has changed, get all vals from manifest
      vars.each do |p|
        val = @resource[p]
        val = @nu.send("default_#{p}") if val == :default
        val = val.to_s if val.is_a?(Symbol)
        attrs[p] = val
      end
    end
    return if attrs.empty?
    @nu.nssa_set(attrs)
  end
```

On the node utils side you'll need to update your yaml (remove the <rm> tag) and then fix route-map in nssa_set, i.e.:

```
+     hash[:route_map] = "route-map #{hash[:route_map]}" unless
+        hash[:route_map].to_s.empty?
       hash.keys.each do |k|
         @set_args[k] = hash[k]
       end
       config_set('ospf_area', 'nssa_set', @set_args)

```